util/file_test.go

```diff
@@ -0,0 +1,45 @@
+package util
+
+import (
+	"testing"
+	"github.com/stretchr/testify/assert"
+	"fmt"
+	"reflect"
+	"github.com/gruntwork-io/boilerplate/errors"
+)
+
+func TestIsTextFile(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		file   string
+		isText bool
+	}{
+		{"binary-file.jpg", false},
+		{"binary-file.png", false},
+		{"binary-file.pdf", false},
+		{"binary-file.zip", false},
+		{"binary-file", false},
+		{"empty-file", false},
+		{"text-file.html", true},
+		{"text-file.js", true},
+		{"text-file.txt", true},
+		{"text-file.md", true},
+		{"text-file.tf", true},
+	}
+
+	for _, testCase := range testCases {
+		actual, err := IsTextFile(fmt.Sprintf("../test-fixtures/util-test/is-text-file/%s", testCase.file))
+
+		assert.Nil(t, err)
+		assert.Equal(t, testCase.isText, actual, "Incorrect classification for %s", testCase.file)
+	}
+}
```

Comment:
I suppose this test will only test the `guessMimeTypeUsingFileCommand` since we run this on Darwin (Mac OS X) and and linux in CircleCI. Since you said the other doesn't work so well, I guess that means it won't work that well in Windows. Is the only consequence of that, that we'll run binary data through the template parser which will slow things down, or something additional to worry about?